src/one/nio/serial/JsonReader.java

```diff
@@ -193,23 +199,33 @@ public Object readNull() throws IOException {
     }
 
     public String readString() throws IOException {
-        StringBuilder sb = new StringBuilder();
-        expect('\"', "Expected string");
-        while (next >= 0 && next != '\"') {
-            int b = read();
-            if ((b & 0x80) == 0) {
-                sb.append(b == '\\' ? readEscapeChar() : (char) b);
-            } else if ((b & 0xe0) == 0xc0) {
-                sb.append((char) ((b & 0x1f) << 6 | (read() & 0x3f)));
-            } else if ((b & 0xf0) == 0xe0) {
-                sb.append((char) ((b & 0x0f) << 12 | (read() & 0x3f) << 6 | (read() & 0x3f)));
-            } else {
-                int v = (b & 0x07) << 18 | (read() & 0x3f) << 12 | (read() & 0x3f) << 6 | (read() & 0x3f);
-                sb.append((char) (0xd800 | (v - 0x10000) >>> 10)).append((char) (0xdc00 | (v & 0x3ff)));
-            }
+        switch (next) {
+            case '\"':
+                StringBuilder sb = new StringBuilder();
+                read();
+                while (next >= 0 && next != '\"') {
+                    int b = read();
+                    if ((b & 0x80) == 0) {
+                        sb.append(b == '\\' ? readEscapeChar() : (char) b);
+                    } else if ((b & 0xe0) == 0xc0) {
+                        sb.append((char) ((b & 0x1f) << 6 | (read() & 0x3f)));
+                    } else if ((b & 0xf0) == 0xe0) {
+                        sb.append((char) ((b & 0x0f) << 12 | (read() & 0x3f) << 6 | (read() & 0x3f)));
+                    } else {
+                        int v = (b & 0x07) << 18 | (read() & 0x3f) << 12 | (read() & 0x3f) << 6 | (read() & 0x3f);
+                        sb.append((char) (0xd800 | (v - 0x10000) >>> 10)).append((char) (0xdc00 | (v & 0x3ff)));
+                    }
+                }
+                expect('\"', "Unexpected end of string");
+                return sb.toString();
+            case 'f':
```

Comment:
I am going for maximum interoperability. Yes, for example GSON will seamlessly parse integer and boolean json input to a String field. It will also try to parse and assign number and boolean fields from string json input.

I might have mention in the comment that not an issue if doing one-nio json to one-nio json. But a potential problem if going with 3rd party json to one-nio json communication. I actually have this case in production where a number field is serialized as string/text not primitive number in the json.